File: bojaxz-cs50-problems-2020-x-sentimental-readability/readability.py
from cs50 import get_string
from string import punctuation

# request input text from the user
text = get_string("Text:")

# define variables for letters, words, sentences
l = 0
w = 0
s = 0

# for loop to count letters

# count sentences by defining a sentence end then counting the number of sentences when seeing a sentence end
seen_end = False
sentence_end = {'?', '!', '.'}
for c in text:
    if c in sentence_end:
        if not seen_end:
            seen_end = True
            s += 1
        continue
    seen_end = False


# Remove puncation to make it easier to count letters in text. Can use len(text) - " "
text = ''.join(ch for ch in text if ch not in punctuation)
for char in text:
    if char != " ":
        l += 1

# use .split() to split words of string. Then count number of words as w
words = text.split()
w = len(words)

# define big L and big S
L = (l * 100) / w
S = (s * 100) / w

# equation for coleman-liau index for readability score
index = int((0.0588 * L - 0.296 * S - 15.8) + 0.5)

# print the resulting grade level from the calculated Coleman-Liau Index using index
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: bojaxz-cs50-problems-2020-x-readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int l;
int w;
int s;


int main(void)
{
    //get input text from user
    string text = get_string("What is the text?: \n");

    //set length of text
    int len = strlen(text);

    //add one if text starts with alnum
    if (isalnum(text[0]))
    {
        w = 1;
    }
    //count number of letters

    for (int i = 0; i < len; i++)
    {
        //count letters
        if (isalpha(text[i]))
        {
            l++;
        }
        //count words
        if (i < len - 1 && isspace(text[i]) && (isalnum(text[i + 1]) || text[i + 1] == '"'))
        {
            w++;
        }
        //count sentances
        if (i > 0 && (text[i] == '!' || text[i] == '?' || text[i] == '.') && isalnum(text[i - 1]))
        {
            s++;
        }
    }
    //calculate Coleman-Liau Index
    double grade = (0.0588 * l * 100 / w - 0.296 * s * 100 / w) - 15.8;
    // int grade = round(graded);
    
    // debugger
    //printf("Letters: %i\n Words: %i\n Sentences: %i\n", l, w, s);

    //return grade
    if (grade <= 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade < 16)
    {
        printf("Grade %.0f\n", round(grade));
        //printf("Grade %f\n", grade);
    }
    else
    {
        printf("Grade 16+\n");
    }
}